app/javascript/src/javascripts/utility/filter_util.js

```diff
@@ -0,0 +1,159 @@
+  status: (token, post) => post.flags.indexOf(token.value) >= 0,
+  rating: (token, post) => post.rating === token.value,
+  type: (token, post) => post.file_ext === token.value,
+
+  width: (token, post) => FilterUtils.compare(post.width, token),
+  height: (token, post) => FilterUtils.compare(post.height, token),
+  filesize: (token, post) => FilterUtils.compare(post.size, token),
+
+  score: (token, post) => FilterUtils.compare(post.score, token),
+  favcount: (token, post) => FilterUtils.compare(post.fav_count, token),
+  fav: (token, post) => post.is_favorited,
+
+  uploader: (token, post) => FilterUtils.FilterTests.user(token, post),
+  user: (token, post) => {
+    // Funky userid: alternative
+    if (token.value.startsWith("!"))
+      return post.uploader_id === parseInt(token.value.substring(1));
+    return post.uploader === token.value;
+  },
+  userid: (token, post) => FilterUtils.compare(post.uploader_id, token),
+  username: (token, post) => post.uploader === token.value,
+
+  pool: (token, post) => post.pools.includes(parseInt(token.value) || 0),
+};
+
+/** Array of supported metatags. */
+const FilterTypes = Object.keys(FilterUtils.FilterTests);
+
+/**
+ * Returns the filter type based on the metatag present in the input.
+ * If none can be found, assumes that this is a regular tag instead.
+ * @param {string} input Input string, should be lower case.
+ * @returns
+ */
+FilterUtils.getFilterType = (input) => {
+  input = input.split(":");
+  if (input.length == 1) return "tag";
+  input = input[0];
+
+  for (const key of FilterTypes)
+    if (input == key) return key;
+  return "tag";
+};
+
+/**
+ * Some people are incredibly strange and put comparisons backwards.
+ * This makes sure that they get normalized regardless.
+ */
+const ComparisonTable = Object.entries({
+  "<": "<",
+  ">": ">",
+  "<=": "<=",
+  ">=": ">=",
+  "=<": "<=",
+  "=>": ">=",
+  "=": "=",
+  "==": "=",
+});
+
+/**
+ * Normalize the comparison type
+ * @param {string} input Comparison string
+ * @returns Normalized comparison string
+ */
+FilterUtils.getComparison = (input) => {
+  if (input.indexOf("..") != -1) return "..";
+  for (const [key, comparison] of ComparisonTable)
+    if (input.startsWith(key)) return comparison;
+  return "=";
+};
+
+/**
+ * Compare the provided value with the one listed in the token
+ * @param {number} a Value to match against
+ * @param {FilterToken} token Token to compare to
+ * @returns true if the provided values pass the specified comparison type
+ */
+FilterUtils.compare = (a, token) => {
+  switch (token.comparison) {
+    case "=":
+      return a == parseFloat(token.value);
+    case "<":
+      return a < parseFloat(token.value);
+    case "<=":
+      return a <= parseFloat(token.value);
+    case ">":
+      return a > parseFloat(token.value);
+    case ">=":
+      return a >= parseFloat(token.value);
+    case "..": {
+      const parts = token.value.split("..");
+      if (parts.length !== 2) return false;
```

Comment:
Within the site only one side needs to be provided, this acts like `>=` / `<=`

`..20` = <=20
`20..` = >=20

https://github.com/e621ng/e621ng/blob/17b99821e3f31ea640a70d51a9631e8584fc11e2/app/logical/parse_value.rb#L40-L41
https://github.com/e621ng/e621ng/blob/17b99821e3f31ea640a70d51a9631e8584fc11e2/app/logical/parse_value.rb#L49-L50

Reply:
That is a silly format.